feat: Added support for customizing rollingUpdate strategy

yaguirre opened the pull request:
This Solves the GitHub Issue #117 

✅ Addresses the core problem: Users can now customize maxSurge and maxUnavailable parameters to prevent rollout failures with HPA and low replica counts
✅ Backward compatible: Existing deployments will continue working with the same behavior (defaults to 1/1)
✅ Flexible configuration: Users can now set custom values in their values.yaml:
yaml
✅ Safe implementation: Only applies to RollingUpdate strategy, preventing errors with Recreate strategy


**How to test**
```
helm template test-release ./harness-delegate-ng --set accountId=test123 --set delegateToken=testtoken123 --set "rollingUpdate.maxSurge=2,rollingUpdate.maxUnavailable=25%" | grep -A 10 "strategy:"
```

**Expected Output**
```
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 2
      maxUnavailable: 25%
  template:
    metadata:
      annotations:
        prometheus.io/path: /api/metrics
        prometheus.io/port: "3460"
        prometheus.io/scrape: "true"

mdmilic commented:
Hi @yaguirre thank you for your contribution!

According to [K8S documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable) the default value for both the fields is actually 25%, not 1. 
Could you please make a change such that unless it's explicitly set, there is no default value from values.yaml and it's not configured in the deployment (so that it takes whatever default K8S provides at that time). 

Thanks,
Marko

yaguirre commented:
> Hi @yaguirre thank you for your contribution!
> 
> According to [K8S documentation](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable) the default value for both the fields is actually 25%, not 1. Could you please make a change such that unless it's explicitly set, there is no default value from values.yaml and it's not configured in the deployment (so that it takes whatever default K8S provides at that time).
> 
> Thanks, Marko

Hi @mdmilic really appreciate your reply.

You’re absolutely right — per the K8s docs the defaults for maxSurge and maxUnavailable are 25% each

I’ve implemented the suggested changes:
- Removed defaults from values.yaml
- Updated the deployment template to only set maxSurge/maxUnavailable when explicitly provided

Please take another look when you have a moment!
